renderlib/gesture/gesture.cpp

```diff
@@ -429,6 +429,32 @@ Gesture::Graphics::RenderBuffer::create(glm::ivec2 resolution, int samples)
   return status;
 }
 
+void
+Gesture::Graphics::SelectionBuffer::clear()
+{
+  // Backup
+  GLenum last_framebuffer;
+  glGetIntegerv(GL_DRAW_FRAMEBUFFER_BINDING, (GLint*)&last_framebuffer);
+  GLfloat last_clear_color[4];
+  glGetFloatv(GL_COLOR_CLEAR_VALUE, last_clear_color);
+
+  // Render to texture
+  glBindFramebuffer(GL_FRAMEBUFFER, frameBuffer);
+  {
+    glViewport(0, 0, resolution.x, resolution.y);
+    uint32_t clearcode = Gesture::Graphics::SelectionBuffer::k_noSelectionCode;
+    glClearColor(((clearcode >> 0) & 0xFF) / 255.0,
+                 ((clearcode >> 8) & 0xFF) / 255.0,
+                 ((clearcode >> 16) & 0xFF) / 255.0,
+                 ((clearcode >> 24) & 0xFF) / 255.0);
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
```

Comment:
Puzzling through this, do you need to specify both `GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT` because the frameBuffer uses color values? Where is depth being used?

Reply:
generally, when you create a framebuffer object you bind both a color buffer and a depth buffer to it. The pick buffer is no different - we want to use depth testing to allow pickable objects to occlude each other just as if we were rendering.  So it makes sense to clear the depth buffer before rendering to it.

Reply:
Gotcha, thank you!